examples/nlp/Tweet-classification-using-TFDF.py

```diff
@@ -0,0 +1,207 @@
+print(f"Training dataset shape: {df.shape}")
+
+# Shuffling training data
+df_shuffled = df.sample(frac=1, random_state=42)
+# Dropping id, keyword and location columns as these columns consists of mostly nan values
+# we will be using only text and target columns
+df_shuffled.drop(["id", "keyword", "location"], axis=1, inplace=True)
+df_shuffled.reset_index(inplace=True, drop=True)
+print(df_shuffled.head())
+
+print(df_shuffled.info())
+
+# Printing total number of Disaster and non-Disaster tweets
+print(
+f"Total Number of disaster and non-disaster tweets\n{df_shuffled.target.value_counts()}"
+)
+
+# Viewing 5 records from training data
+for ind, counter in enumerate(df_shuffled.index):
+    print(f"Target : {df_shuffled['target'][ind]}")
+    print(f"Text : {df_shuffled['text'][ind]}")
+    if counter == 5:
+        break
+
+# Splitting dataset into train and test
+test_df = df_shuffled.sample(frac=0.2, random_state=42)
+train_df = df_shuffled.drop(test_df.index)
+
+print(f"Using {len(train_df)} samples for training and {len(test_df)} for validation")
+
+"""
+## Convert data to tf.Dataset
+"""
+
+
+def create_dataset(dataframe):
+    df = dataframe.copy()
+    dataset = tf.data.Dataset.from_tensor_slices(
+        (df["text"].to_numpy(), df["target"].to_numpy())
+    )
+    dataset = dataset.batch(100)
+    dataset = dataset.prefetch(tf.data.AUTOTUNE)
+    return dataset
+
+
+train_ds = create_dataset(train_df)
+test_ds = create_dataset(test_df)
+
+"""
+## Downloading Pre-trained Embeddings
+
+The Universal Sentence Encoder encodes text into high-dimensional vectors that can be
+used for text classification, semantic similarity, clustering and other natural language
+tasks.It is trained on a variety of data sources and a variety of tasks. The input is
+variable length English text and the output is a 512 dimensional vector.
+
+To understand better about pre-trained embeddings : 
+[Universal Sentence Encoder](https://tfhub.dev/google/universal-sentence-encoder/4)
+
+
+"""
+
+sentence_encoder_layer = hub.KerasLayer(
+    "https://tfhub.dev/google/universal-sentence-encoder/4"
+)
+
+"""
+## Build a model
+
+First, we develop a preprocessor model that takes the inputs and passes it through
+sentence encoder layer to generate embedding, which produces a 512 dimensional vector for
+each sentence.
+
+The Preprocessor is then passed to the GradientBoostedTreesModel which will be used for
+predicting disaster tweets.
+
+
+"""
+
+inputs = layers.Input(shape=(), dtype=tf.string)
+# Passing sentences to sentence_encoder_layer
+outputs = sentence_encoder_layer(inputs)
+
+preprocessor = keras.Model(inputs=inputs, outputs=outputs)
+model = tfdf.keras.GradientBoostedTreesModel(preprocessing=preprocessor)
+
+"""
+## Train the model
+"""
+
+# Compiling model
+model.compile(metrics=["Accuracy", "Recall", "Precision", "AUC"])
+# Here we do not specify epochs as, TF-DF trains exactly one epoch of the dataset
+model.fit(x=train_ds)
```

Comment:
We need to monitor validation accuracy on a validation set.

Reply:
As per the documentation, TFDF does not require a validation set [docs](https://www.tensorflow.org/decision_forests/migration#validation_dataset)